### PR TITLE
Adding MotionControllerModel project

### PR DESCRIPTION
**Pull request by keveleigh**
This allows us to get the motion controller models in the editor's play mode.

### CODE REVIEW

File: MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                        return FALSE;
+                    }
+
+                    // Wait until the model call finishes.
+                    while (modelOperation->Status == AsyncStatus::Started)
+                    {
+                        std::chrono::milliseconds milis(1);
+                        std::this_thread::sleep_for(milis);
+                    }
+
```
Comment (fieldsJacksonG) [inline MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: Return false if modelOperation->Status != Completed

Comment (fieldsJacksonG) [reply MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: (Canceled or Error)

File: MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                {
+                    // Start getting the renderable model stream asynchronously.
+                    auto modelOperation = sources->GetAt(i)->Source->Controller->TryGetRenderableModelAsync();
+
+                    if (modelOperation == nullptr)
+                    {
+                        return FALSE;
+                    }
+
+                    // Wait until the model call finishes.
```
Comment (fieldsJacksonG) [inline MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: Can you use ppl for these IAsyncOperations?

Comment (fieldsJacksonG) [reply MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: https://msdn.microsoft.com/en-us/library/dd492418.aspx

File: MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+using namespace Microsoft::WRL;
+using namespace Microsoft::WRL::Wrappers;
+using namespace Windows::Foundation;
+using namespace Windows::Perception;
+using namespace Windows::Storage::Streams;
+using namespace Windows::UI::Input::Spatial;
+
+extern "C"
+{
+    EXTERN_C __declspec(dllexport) BOOL TryGetMotionControllerModel(_In_ UINT32 controllerId, _Out_ UINT32* outputSize, _Outptr_result_bytebuffer_(outputSize) BYTE** outputBuffer)
```
Comment (fieldsJacksonG) [inline MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: UINT32&, then you don't need to dereference outputSize below

Comment (fieldsJacksonG) [reply MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: Same for BYTE*& outputBuffer

File: MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            // Get the currently detected spatial sources.
+            auto sources = spatialInteractionManager->GetDetectedSourcesAtTimestamp(perceptionTimestamp);
+
+            // Iterate through the detected sources and check against the controller ID that was passed in.
+            for (unsigned int i = 0; i < sources->Size; i++)
+            {
+                if (sources->GetAt(i)->Source->Id == controllerId)
+                {
+                    // Start getting the renderable model stream asynchronously.
+                    auto modelOperation = sources->GetAt(i)->Source->Controller->TryGetRenderableModelAsync();
```
Comment (fieldsJacksonG) [inline MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: Version check, in case this DLL gets included in a HoloLens project

Comment (fieldsJacksonG) [reply MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: or a try catch

File: MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+#include "stdafx.h"
+
+using namespace Microsoft::WRL;
+using namespace Microsoft::WRL::Wrappers;
+using namespace Windows::Foundation;
+using namespace Windows::Perception;
+using namespace Windows::Storage::Streams;
+using namespace Windows::UI::Input::Spatial;
+
+extern "C"
```
Comment (fieldsJacksonG) [inline MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: extern "C" black and EXTERN_C macro are redundant

File: MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp (hunk 1 above)
Comment (fieldsJacksonG) [reply MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: I would expect an error case to be either not_completed or an exception (depending on the error!)
https://msdn.microsoft.com/en-us/library/dd997692.aspx

File: MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        {
+            if (sources->GetAt(i)->Source->Id == controllerId)
+            {
+                // Start getting the renderable model stream asynchronously.
+                task<IRandomAccessStreamWithContentType^> modelOperation;
+                try
+                {
+                    modelOperation = create_task(sources->GetAt(i)->Source->Controller->TryGetRenderableModelAsync());
+                }
+                catch (std::string e)
```
Comment (fieldsJacksonG) [inline MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: catch(...)

File: MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                status = readOperation.wait();
+
+                if (status != task_status::completed)
+                {
+                    return FALSE;
+                }
+
+                // Now, create a DataReader from the buffer, which can then transfer the bytes into a byte array.
+                DataReader^ reader = DataReader::FromBuffer(buffer);
+                BYTE* pixels = new BYTE[buffer->Length];
```
Comment (fieldsJacksonG) [inline MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: outputBuffer = new BYTE ?

Comment (fieldsJacksonG) [reply MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: Then you don't need an intermediate container

File: MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+using namespace Concurrency;
+using namespace Microsoft::WRL;
+using namespace Microsoft::WRL::Wrappers;
+using namespace Windows::Foundation;
+using namespace Windows::Perception;
+using namespace Windows::Storage::Streams;
+using namespace Windows::UI::Input::Spatial;
+
+EXTERN_C __declspec(dllexport) BOOL TryGetMotionControllerModel(_In_ UINT32 controllerId, _Out_ UINT32& outputSize, _Outptr_result_bytebuffer_(outputSize) BYTE*& outputBuffer)
```
Comment (fieldsJacksonG) [inline MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: should outputSize be a LONGLONG?

Comment (keveleigh) [reply MotionControllerModel/MotionControllerModel/MotionControllerModel.cpp]: image

**Pull request merged**